MSAL/src/MSALAccount+Internal.h

```diff
@@ -25,61 +25,55 @@
 //
 //------------------------------------------------------------------------------
 
-#import <Foundation/Foundation.h>
+#import "MSALAccount.h"
 
-@interface MSALUser : NSObject <NSCopying>
+@class MSIDAccountIdentifier;
+@class MSIDAADV2IdTokenClaims;
+@class MSIDClientInfo;
+@class MSIDAccount;
+@class MSALAccountId;
 
-/*!
- The displayable value in UserPrincipleName(UPN) format. Can be nil if not returned from the service.
- */
-@property (readonly) NSString *displayableId;
+@interface MSALAccount ()
 
-/*!
- The given name of the user. Can be nil if not returned by the service.
- */
-@property (readonly) NSString *name;
+@property (nonatomic) MSIDAccountIdentifier *lookupAccountIdentifier;
 
-/*!
- The identity provider of the user authenticated. Can be nil if not returned by the service.
+/* TODO: These properties will be public once we agree on having an account per tenant.
+   For now, will keep them here.
  */
-@property (readonly) NSString *identityProvider;
 
 /*!
- Unique identifier of the user. Can be nil if not returned by the service.
+ Account identifier for the target tenant
  */
-@property (readonly) NSString *uid;
+@property (nonatomic) MSALAccountId *localAccountId;
 
 /*!
- Unique tenant identifier of the user. Can be nil if not returned by the service.
+ The displayable name of the account. Can be nil if not returned by the service.
  */
-@property (readonly) NSString *utid;
+@property (nonatomic) NSString *name;
 
-/*!
- Host part of the authority string used for authentication.
- */
-@property (readonly) NSString *environment;
 
 /*!
- Initialize a MSALUser with given information
- 
+ Initialize an MSALAccount with given information
+
  @param  displayableId       The displayable value in UserPrincipleName(UPN) format
  @param  name                The given name of the user
- @param  identityProvider    The identity provider of the user authenticated
- @param  uid                 Unique identifier of the user
- @param  utid                Unique tenant identifier of the user
+ @param  homeAccountId       Unique identifier of the account in the home directory
+ @param  localAccountId      Unique identifier of the account in the signed in directory.
  @param  environment         Host part of the authority string
+ @param  tenantId            An identifier for the tenant that the account was acquired from
  */
 - (id)initWithDisplayableId:(NSString *)displayableId
                        name:(NSString *)name
-           identityProvider:(NSString *)identityProvider
-                        uid:(NSString *)uid
-                       utid:(NSString *)utid
-                environment:(NSString *)environment;
+              homeAccountId:(NSString *)homeAccountId
```

Comment:
Question regarding parameters above this line.
Confused between displayableId and name, as I just saw name = displayable name above.

Reply:
Renamed displayableId to username to be clear. Name is based on the id_token's claim "name", so I'd prefer to just leave it that way. Please let me know if you disagree :)